js/12/script.js

```diff
@@ -0,0 +1,168 @@
+// Если он передает неправильный тип гамбургера, например, или неправильный вид добавки,
+// должно выбрасываться исключение (ошибка не должна молча игнорироваться).
+Hamburger.SIZE_SMALL = {price: 50, calories: 20};
+Hamburger.SIZE_LARGE = {price: 100, calories: 40};
+Hamburger.STUFFING_CHEESE = {price: 10, calories: 20};
+Hamburger.STUFFING_SALAD = {price: 20, calories: 5};
+Hamburger.STUFFING_POTATO = {price: 15, calories: 10};
+Hamburger.TOPPING_MAYO = {price: 20, calories: 5};
+Hamburger.TOPPING_SPICE = {price: 15, calories: 0};
+
+function Hamburger(size, stuffing) {
+    try {
+        this.size = size || Hamburger.SIZE_SMALL;
+        this.stuffing = stuffing || Hamburger.STUFFING_CHEESE;
+
+        try {
+            if ((typeof size !== "object" || typeof size === undefined)) {
+                throw new HamburgerException("Invalid parameters. Must be known constant. Default value for size - small");
+            }
+        }
+        catch (e) {
+            console.log(e.name + " " + e.message);
+        }
+
+        try {
+            if (typeof stuffing !== "object" || typeof stuffing === undefined) {
+                throw new HamburgerException("Invalid parameters. Must be known constant. Default value for stuffing - cheese");
+            }
+        }
+        catch (e) {
+            console.log(e.name + " " + e.message);
+        }
+    }
+    catch (e) {
+    }
+}
+
+var toppingArr = [];
+
+Hamburger.prototype.addTopping = function (topping) {
+    try {
+        if (topping === undefined || typeof topping !== "object") {
+            throw new HamburgerException("Invalid parameter. Must be known constant");
+        }
+    }
+    catch (e) {
+        console.log(e.name + " " + e.message);
+    }
+
+    this.topping = topping;
+    var ex = false;
+
+    if (toppingArr[0] === undefined) {
+        toppingArr.push(this.topping);
+    }
+    else {
+        for (var i = 0; i < toppingArr.length; ++i) {
+            if (this.topping === toppingArr[i]) {
+                console.log("Topping exists");
+                ex = true;
+            }
+        }
+        if (!ex) {
+            toppingArr.push(this.topping);
+        }
+    }
+    return toppingArr;
+}
+
+Hamburger.prototype.removeTopping = function (topping) {
+    try {
+        if (topping === undefined || typeof topping !== "object") {
+            throw new HamburgerException("Invalid parameter. Must be known constant");
+        }
+    }
+    catch (e) {
+        console.log(e.name + " " + e.message);
+    }
+
+    if (this.topping === toppingArr[0]) {
+        toppingArr.shift();
+    }
+    else {
+        toppingArr.pop();
+    }
+
+    return toppingArr;
+}
+
+Hamburger.prototype.getToppings = function () {
+    return toppingArr;
+}
+
+Hamburger.prototype.getSize = function () {
+    if (this.size === Hamburger.SIZE_SMALL) {
+        return Hamburger.SIZE_SMALL;
+    }
+
+    return Hamburger.SIZE_LARGE;
```

Comment:
return this.size === Hamburger.SIZE_SMALL ? Hamburger.SIZE_SMALL : Hamburger.SIZE_LARGE;